tests/test_graph.py

```diff
@@ -12,28 +12,34 @@
 name_sets = [['python2'],
              ['kivy']]
 bootstraps = [None,
-              Bootstrap.get_bootstrap('pygame', ctx),
               Bootstrap.get_bootstrap('sdl2', ctx)]
 valid_combinations = list(product(name_sets, bootstraps))
 valid_combinations.extend(
     [(['python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx)),
-     (['kivy', 'python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx))])
+     (['kivy', 'python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx)),
+     (['flask'], Bootstrap.get_bootstrap('webview', ctx))])
 invalid_combinations = [[['python2', 'python3crystax'], None]]
 
 
 @pytest.mark.parametrize('names,bootstrap', valid_combinations)
 def test_valid_recipe_order_and_bootstrap(names, bootstrap):
-    get_recipe_order_and_bootstrap(ctx, names, bootstrap)
+    try:
+        get_recipe_order_and_bootstrap(ctx, names, bootstrap)
+    except Exception as e:
+        # pytest output is too terse, so do something better here:
+        print(
+            "test_valid_recipe_order_and_bootstrap FAILED " +
+            "with these specific parameters: " +
+            str((ctx, names, bootstrap))
+        )
+        raise e
```

Comment:
That test looks a bit confusing now isn't it? Is this all just to make the failing test output easier to read on crash?
I would prefer to avoid that, specially being that broad catching `Exception` what if the code as a `SyntaxError` for instance...

Reply:
As you wish, it just feel not right/hacky to have something like this in the unit test and I feel like it could confuse more than it would help. But that's maybe because I didn't see what you have seen.
You can ignore my comment then

Reply:
@AndreMiras I tried to clarify it some more. basically if the `names=...` set is too big, pytest just cuts off the members which is very annoying trying to find out what was problematic about the dependency combinations for which the test failed. I pretty much just added this to make sure one can always see the full dependency set tested even if pytest decided to cut it off

Reply:
Thanks for adding that comment, that may help future maintainers to understand why we have something weird here.
So you said if `names=...` is too long. But I don't see it too long in any of the tests here. Plus when a test fail dev put a breakpoint before the exception is raised anyway. So why don't we keep things simple?
Really it's hard for me to see the value it provides, it just feels like confusion. I tried without the try/catch block and the code just runs the same on conflicting case. I've only created a small `names` list, but that's what are test have anyway isn't it?

Reply:
> But I don't see it too long in any of the tests here.

Oh it is, I added it because it was cut off and that wasted my time. I don't like it when test output wastes my time :smile: but same applies here, if you all vote against it I remove it. But I like my tests to always produce useful output, not only 90% of the time

>  I've only created a small `names` list, but that's what are test have anyway isn't it?

There's at least one that has a longer one, otherwise it wouldn't have been cut off :woman_shrugging: it certainly wasn't due to a *hypothetical* cut off, if that is what you mean

Reply:
yep I was just curious sorry, I definitely looked it up and didn't find it.
I do also think not wasting time is very import just as code readability.
I won't waste yours more you already did an awesome job, so leave it. But to be clear, yep it definitely not convinced that try/catch provide value but rather the opposite, at least to me

Reply:
@AndreMiras thinking about this more, I think you're right. I removed it, it was only for one of the pairs and a pytest output option maybe would even have fixed it...? it seems odd to hack this in for a single corner case. I removed it again :+1: 

hope it's okay to resolve this since hopefully this should be cleared up now :heart:

Reply:
Thank you I can sleep tight now! :relaxed: